### PR TITLE
Lua: fix global module loading

### PR DESCRIPTION
**Pull request by tarleb**
Prevent issues if Lua is misconfigured on the system.

### CODE REVIEW

File: src/Text/Pandoc/Lua/Init.hs [hunk 1]
```diff
@@ -78,23 +78,43 @@ initLuaState = do
...
             Lua.pop 1  -- ignore error message
             -- Try loading via the normal package loading mechanism.
             Lua.getglobal "require"
             Lua.pushName pkgname
             Lua.call 1 1  -- Throws an exception if loading failed again!
+            return False
 
         -- Module on top of stack. Register as global
         Lua.setglobal pkgname
+        return usedInternal
```
Comment (jgm) [inline src/Text/Pandoc/Lua/Init.hs:107]: How about
```
return (usedInternal, pkgname)
```
and then avoid the `zip` on line 112 and the `_` on 113?

Comment (jgm) [reply src/Text/Pandoc/Lua/Init.hs:107]: Or, perhaps even clearer, return `Just pkgname` instead of True and `Nothing` instead of False.

**Comment by jgm** [comment]
I think I prefer this solution to the "no re" solution, now that I understand it.
Sorry, tests failed because of something I've since fixed, so you'll want to rebase.

File: src/Text/Pandoc/Lua/Init.hs (hunk 1 above)
Comment (tarleb) [reply src/Text/Pandoc/Lua/Init.hs]: Much nicer. :+1:

**Comment by tarleb** [comment]
Updated and rebased. Tests pass, so I hope it's ok if I just merge.

**Comment by jgm** [comment]
I pulled this in and I still get a test failure (this is on another machine):
```
  Lua
    Lua filters
      global modules
        module 'lpeg' is available via `require`:                                                                                   FAIL (0.40s)
          test/Tests/Lua.hs:225:
          expected: OK
           but got: ErrRun
          Use -p '/module '\''lpeg'\'' is available via `require`/' to rerun this test only.
        module 're' is available via `require`:                                                                                     FAIL (0.38s)
          test/Tests/Lua.hs:229:
          expected: OK
           but got: ErrRun
          Use -p '/module '\''re'\'' is available via `require`/' to rerun this test only.

2 out of 2947 tests failed (157.62s)
```
Note that on this machine (as on the one I was using before) I have lpeg installed but it's in lua 5.4.
If I reset LUA_CPATH and run the test suite again,  it passes.
Is this expected?  Or is it supposed to fall back to the in-built lpeg if there are problems loading the system one?  (I know we discussed this but can't remember what was decided.)

If it's expected behavior, I will work around it by having the test suite reset LUA_CPATH, so that tests can succeed regardless of the test machine's lua setup.

**Comment by jgm** [comment]
I've changed the two failed tests so they reset the lua path and cpath before doing `require`. This ensures that what the test is testing is the same on every system, regardless of its Lua setup. @tarleb let me know if this doesn't seem like a good idea.

**Comment by tarleb** [comment]
Thanks, that's the exact change I was planning to make after reading your first message.

Probably no longer relevant, but the version collisions can be worked around by setting `LUA_CPATH_5_3`; Lua should ignore `LUA_CPATH` if a version specific env var is set.

**Comment by jgm** [comment]
So there's a commit that says:

>   Make loading of global LPeg modules more robust.
      Ignore errors if the normal package mechanism failed; this not only
      covers the case of modules being unavailable on the system, but also
      works if the modules are present, but fail to load for some reason.
      This makes the built-in package version a true fallback.

Has that behavior been superseded?  Or should it work that way and fall back to the inbuilt lpeg in this case?

**Comment by tarleb** [comment]
That is indeed outdated now. I tried to describe the current behavior in `lua-filters.md`, but it may be a little too terse:

> Pandoc uses a built-int version of the library,
> unless it has been configured by the package maintainer to
> rely on a system-wide installation.

> Note that the result of `require 'lpeg'` is not necessarily
> equal to this value; the `require` mechanism prefers the
> system's lpeg library over the built-in version.

**Pull request merged**